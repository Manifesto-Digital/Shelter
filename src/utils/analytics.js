export const sendEvent = data => {
  window.dataLayer.push(data);
};
