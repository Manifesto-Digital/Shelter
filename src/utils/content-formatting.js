export const addMarksToContent = content => {
  const markup = content
    .map(block => {
      const { marks, value } = block
      // As there may be multiple marks per piece of content, we need to
      // store the opening and closing tags and then wrap the content
      if (marks && marks.length > 0) {
        let openingMarks = ''
        let closingMarks = ''

        marks.forEach(mark => {
          if (mark.type === 'underline') {
            openingMarks = `<span className="underline">${openingMarks}`
            closingMarks = `${closingMarks}</span>`
          }
          if (mark.type === 'italic') {
            openingMarks = `<em>${openingMarks}`
            closingMarks = `${closingMarks}</em>`
          }
          if (mark.type === 'bold') {
            openingMarks = `<strong>${openingMarks}`
            closingMarks = `${closingMarks}</strong>`
          }
        })
        return `${openingMarks}${value}${closingMarks}`
      }
      return value
    })
    .join('')

  return markup
}
