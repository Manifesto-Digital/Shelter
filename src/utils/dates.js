import dayjs from 'dayjs';

// Format date as string in required format
// Formatting options: https://bit.ly/2R9uVsh
export const dateAsString = (date, format) =>
  dayjs(date)
    .format(format)
    .toString();

export const getWeekDays = () => {
  // Base date is a Monday, this ensures we get the day lsit in order
  const baseDate = new Date(Date.UTC(2019, 0, 7));
  const weekDays = [];
  for (let i = 0; i < 7; i += 1) {
    weekDays.push(baseDate.toLocaleDateString('en-GB', { weekday: 'long' }));
    baseDate.setDate(baseDate.getDate() + 1);
  }
  return weekDays;
};

export const dayOfTheWeek = () => {
  const daysArray = getWeekDays();
  const today = new Date();

  return daysArray[today.getDay() - 1];
};
