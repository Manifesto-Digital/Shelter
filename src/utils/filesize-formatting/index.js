/**
 * Formats integer to human readable filesize
 *
 * @param {int} bytes
 * @param {int} decimalPoint
 */
export const formatFilesize = (bytes, decimalPoint) => {
  if (bytes === undefined) throw new Error('Bytes are required');
  if (bytes < 0) throw new Error('Negative bytes not allowed');
  if (bytes >= 10 ** 27)
    throw new Error(`Bytes too large, must be < 10 ** 27 (${10 ** 27})`);

  if (bytes === 0) return '0 Bytes';

  const k = 1000;
  const dm = decimalPoint || 2;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  const sizeToReturn = `${parseFloat((bytes / k ** i).toFixed(dm))} ${
    sizes[i]
  }`;
  return sizeToReturn;
};
