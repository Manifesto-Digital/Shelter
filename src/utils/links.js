export const getInternalLink = slug => {
  const baseURL = window.location.origin
  return `${baseURL}/${slug}`
}
