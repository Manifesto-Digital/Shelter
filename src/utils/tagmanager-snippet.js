const tagManagerSnippet = () =>
  (function(w, d, s, l, i) {
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    const f = d.getElementsByTagName(s)[0];
    const j = d.createElement(s);
    const dl = l != 'dataLayer' ? `&l=${l}` : ''; // eslint-disable-line eqeqeq
    j.async = true;
    j.src = `https://www.googletagmanager.com/gtm.js?id=${i}${dl}`;
    f.parentNode.insertBefore(j, f);
  })(window, document, 'script', 'dataLayer', 'GTM-WWSW2HH');

export default tagManagerSnippet;
