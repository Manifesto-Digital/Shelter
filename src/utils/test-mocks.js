// Whenever LinkHandler is used, when mounted in enzyme the rendered HTML
// includes a lowercase "l" in the gatsby Link tag. Which causes a warning
// This suppresses that warning for a whole test file
export const hidePascalCaseWarning = () => {
  beforeEach(() => {
    console.error = jest.fn(error => {
      if (
        !error.includes(
          'using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements'
        )
      ) {
        throw new Error(error);
      }
    });
  });

  afterEach(() => {
    console.error.mockClear();
  });
};
