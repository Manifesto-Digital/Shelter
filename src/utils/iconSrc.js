import SpeechBubble from '../assets/svg/icons/comment-light.svg';
import Envelope from '../assets/svg/icons/envelope-light.svg';
import Download from '../assets/svg/icons/download-light.svg';
import Printer from '../assets/svg/icons/print-light.svg';
import OpenBook from '../assets/svg/icons/book-open-light.svg';
import Exclamation from '../assets/svg/icons/exclamation-circle-light.svg';
import Telephone from '../assets/svg/icons/phone-light.svg';
import MapMarker from '../assets/svg/icons/map-marker-alt-light.svg';
import ChippedTimeResults from '../assets/svg/icons/challenge-event/chipped-time-results.svg';
import FinishersMedal from '../assets/svg/icons/challenge-event/finishers-medal.svg';
import FreeBagTransfer from '../assets/svg/icons/challenge-event/free-bag-transfer.svg';
import FreeRacePhotos from '../assets/svg/icons/challenge-event/free-race-photos.svg';
import GoodieBags from '../assets/svg/icons/challenge-event/goodie-bags.svg';
import PostRaceCelebration from '../assets/svg/icons/challenge-event/post-race-celebration.svg';
import SportsMassage from '../assets/svg/icons/challenge-event/sports-massage.svg';
import TechnicalRunningTop from '../assets/svg/icons/challenge-event/technical-running-top.svg';
import ClipboardChecklist from '../assets/svg/icons/clipboard-checklist-light.svg';
import FundingGoalMeter from '../assets/svg/icons/funding-goal-meter-light.svg';
import Lightbulb from '../assets/svg/icons/lightbulb-light.svg';
import MapUnfolded from '../assets/svg/icons/map-unfolded-light.svg';

import { consistentString } from './content-formatting';

const iconList = {
  'speech-bubble': SpeechBubble,
  envelope: Envelope,
  download: Download,
  printer: Printer,
  'open-book': OpenBook,
  'exclamation-mark': Exclamation,
  telephone: Telephone,
  'map-marker': MapMarker,
  'chipped-time-results': ChippedTimeResults,
  'finishers-medal': FinishersMedal,
  'free-bag-transfer': FreeBagTransfer,
  'free-race-photos': FreeRacePhotos,
  'goodie-bags': GoodieBags,
  'post-race-celebration': PostRaceCelebration,
  'sports-massage': SportsMassage,
  'technical-running-top': TechnicalRunningTop,
  'clipboard-checklist': ClipboardChecklist,
  'funding-goal-meter': FundingGoalMeter,
  lightbulb: Lightbulb,
  'map-unfolded': MapUnfolded,
  'no-icon': null,
  default: null,
};

// returns the src of an svg icon from the above icon list
const iconSrc = icon => {
  const iconKey = consistentString(icon);

  if (iconList[iconKey] === undefined) {
    throw new Error(
      'No matching icon was found. Is it set as an option in the CMS? Has it been imported and included in the object above?'
    );
  }

  const chosenIcon = iconList[iconKey] || null;

  if (!chosenIcon) return null;

  return chosenIcon;
};

export default iconSrc;
