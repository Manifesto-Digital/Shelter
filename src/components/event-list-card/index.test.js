import React from 'react';
import { Link } from 'gatsby';
import {
  createFactory,
  createImage,
  createInternalRef,
} from '../../utils/test-factories';
import { hidePascalCaseWarning } from '../../utils/test-mocks';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../__tests__/helpers/index';
import EventListCard from './index';

export const createEventListCard = createFactory({
  slug: 'my-slug',
  event: {
    thumbnailImage: createImage(),
    eventName: 'My Event',
    displayLocation: 'round the corner',
    eventDisplayDate: 'Yesterday',
    eventSystemDate: '2019-02-25',
    link: createInternalRef(),
  },
});

it('renders correctly', () => {
  const mockData = createEventListCard();

  snapshotComponent(<EventListCard data={mockData} />);
});

hidePascalCaseWarning();

it('displays the specified image correctly', () => {
  const mockData = createEventListCard();
  const wrapper = mountWithTheme(<EventListCard data={mockData} />);
  expect(wrapper.find('img').prop('src')).toContain(
    mockData.event.thumbnailImage.file.url
  );
});

it('sets the correct CTA links', () => {
  const mockData = createEventListCard({ slug: 'my-test-slug' });
  const wrapper = mountWithTheme(<EventListCard data={mockData} />);

  expect(
    wrapper
      .find(Link)
      .at(0)
      .prop('to')
  ).toContain('my-test-slug');
});
