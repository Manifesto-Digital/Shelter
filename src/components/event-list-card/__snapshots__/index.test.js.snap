// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c6 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 0;
  opacity: 0;
}

.c0 {
  padding: 20px;
  margin-bottom: 20px;
  background-color: #fff;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  border-radius: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c0:hover {
  background-color: #f7f7f7;
}

.c1 {
  display: none;
}

.c3 {
  padding: 0;
  list-style: none;
}

.c4 {
  font-size: 0.775rem;
  margin: 0;
}

.c5 {
  margin-right: 10px;
  display: inline-block;
}

.c5 svg {
  height: 0.775rem;
  width: 0.775rem;
}

@media (min-width:32em) {
  .c0 {
    padding: 10px;
  }
}

@media (min-width:32em) {
  .c1 {
    display: block;
    -webkit-align-self: start;
    -ms-flex-item-align: start;
    align-self: start;
    width: 50%;
  }
}

@media (min-width:62em) {
  .c1 {
    width: 35%;
  }
}

@media (min-width:32em) {
  .c2 {
    margin-left: 20px;
  }
}

<div
  className="c0"
>
  <img
    alt="Collection hero"
    className="c1"
    src="//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg?w=600"
  />
  <div
    className="c2"
  >
    <h3>
      My Event
    </h3>
    <ul
      className="c3"
    >
      <li
        className="c4"
      >
        <span
          className="isvg loading c5"
        />
        round the corner
      </li>
      <li
        className="c4"
      >
        <span
          className="isvg loading c5"
        />
        Yesterday
      </li>
    </ul>
  </div>
  <Link
    className="c6"
    to="/my-slug"
  >
    My Event
  </Link>
</div>
`;
